Mathlib/Probability/Moments.lean

```diff
@@ -285,6 +286,21 @@ theorem iIndepFun.cgf_sum {X : ι → Ω → ℝ}
   · rw [h_indep.mgf_sum h_meas]
   · exact (mgf_pos (h_int j hj)).ne'
 
+theorem mgf_congr_identDistrib
+    (X : Ω → ℝ) {Ω' : Type*} {m' : MeasurableSpace Ω'} {μ' : Measure Ω'} (X' : Ω' → ℝ)
+    (hident : IdentDistrib X X' μ μ') (t : ℝ) :
+    mgf X μ t = mgf X' μ' t := hident.comp (measurable_const_mul t).exp |>.integral_eq
+
+theorem mgf_sum_of_identDistrib
+    {X : ι → Ω → ℝ}
+    (h_meas : ∀ (i : ι), Measurable (X i))
+    (h_indep : ProbabilityTheory.iIndepFun (fun _ => inferInstance) X μ)
+    (s : Finset ι) (j : s)
```

Comment:
It's better to have `j : ι` + `j ∈ s` rather than ` j : s`
```suggestion
    {s : Finset ι} {j : ι} (hj : j ∈ s)
```
Also I think `hj` should be the last proof obligation in this statement, namely it should come just before `(t : ℝ)`